examples/cd_distillation_cifar10.ipynb

```diff
@@ -0,0 +1,781 @@
+    "import pandas as pd\n",
+    "import os\n",
+    "import tensorflow as tf\n",
+    "from tensorflow.keras.layers import Conv2D, Dense, Flatten, InputLayer\n",
+    "from tensorflow.keras.regularizers import l1\n",
+    "from alibi_detect.cd import KSDrift\n",
+    "from alibi_detect.ad import ModelDistillation\n",
+    "\n",
+    "from alibi_detect.models.resnet import scale_by_instance\n",
+    "from alibi_detect.utils.fetching import fetch_tf_model, fetch_detector\n",
+    "from alibi_detect.utils.prediction import predict_batch\n",
+    "from alibi_detect.utils.saving import save_detector, load_detector\n",
+    "from alibi_detect.datasets import fetch_cifar10c, corruption_types_cifar10c"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load data\n",
+    "\n",
+    "Original CIFAR-10 data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(X_train, y_train), (X_test, y_test) = tf.keras.datasets.cifar10.load_data()\n",
+    "X_train = X_train.astype('float32') / 255\n",
+    "X_test = X_test.astype('float32') / 255\n",
+    "y_train = y_train.astype('int64').reshape(-1,)\n",
+    "y_test = y_test.astype('int64').reshape(-1,)\n",
+    "X_train = scale_by_instance(X_train)\n",
+    "X_test = scale_by_instance(X_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For CIFAR-10-C, we can select from the following corruption types at 5 severity levels:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "['brightness', 'contrast', 'defocus_blur', 'elastic_transform', 'fog', 'frost', 'gaussian_blur', 'gaussian_noise', 'glass_blur', 'impulse_noise', 'jpeg_compression', 'motion_blur', 'pixelate', 'saturate', 'shot_noise', 'snow', 'spatter', 'speckle_noise', 'zoom_blur']\n"
+     ]
+    }
+   ],
+   "source": [
+    "corruptions = corruption_types_cifar10c()\n",
+    "print(corruptions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's pick a subset of the corruptions at corruption level 5. Each corruption type consists of perturbations on all of the original test set images."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "corruption = ['gaussian_noise', 'motion_blur', 'brightness', 'pixelate']\n",
+    "X_corr, y_corr = fetch_cifar10c(corruption=corruption, severity=5, return_X_y=True)\n",
+    "X_corr = X_corr.astype('float32') / 255"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We split the original test set in a reference dataset and a dataset which should not be rejected under the *H<sub>0</sub>* of the K-S test. We also split the corrupted data by corruption type:"
```

Comment:
There is no reference or *H0* dataset created. Both are necessary to test the validity of the detector. If e.g. also drift occurs vs. *H0* the results on the corrupted data carry little value.

Reply:
The reference set is created below, when the KS detector is initialized. I think it's clear from the plots at the end that a set with no contamination would not be rejected under the null hypothesis. The comment is misplaced however, I removed it